lib/spack/spack/test/compilers/flags.py

```diff
@@ -582,165 +571,6 @@ def test_xl_r_flags():
-    ('4.4.7\n', '4.4.7'),
-    ('7\n', '7')
-])
-def test_gcc_version_detection(version_str, expected_version):
-    version = spack.compilers.gcc.Gcc.extract_version_from_output(version_str)
-    assert version == expected_version
-
-
-@pytest.mark.parametrize('version_str,expected_version', [
-    ('icpc (ICC) 12.1.5 20120612\n'
-     'Copyright (C) 1985-2012 Intel Corporation.  All rights reserved.\n',
-     '12.1.5'),
-    ('ifort (IFORT) 12.1.5 20120612\n'
-     'Copyright (C) 1985-2012 Intel Corporation.  All rights reserved.\n',
-     '12.1.5')
-])
-def test_intel_version_detection(version_str, expected_version):
-    version = compilers.intel.Intel.extract_version_from_output(version_str)
-    assert version == expected_version
-
-
-@pytest.mark.parametrize('version_str,expected_version', [
-    ('NAG Fortran Compiler Release 6.0(Hibiya) Build 1037\n'
-     'Product NPL6A60NA for x86-64 Linux\n', '6.0')
-])
-def test_nag_version_detection(version_str, expected_version):
-    version = spack.compilers.nag.Nag.extract_version_from_output(version_str)
-    assert version == expected_version
-
-
-@pytest.mark.parametrize('version_str,expected_version', [
-    # Output on x86-64
-    ('pgcc 15.10-0 64-bit target on x86-64 Linux -tp sandybridge\n'
-     'The Portland Group - PGI Compilers and Tools\n'
-     'Copyright (c) 2015, NVIDIA CORPORATION.  All rights reserved.\n',
-     '15.10'),
-    # Output on PowerPC
-    ('pgcc 17.4-0 linuxpower target on Linuxpower\n'
-     'PGI Compilers and Tools\n'
-     'Copyright (c) 2017, NVIDIA CORPORATION.  All rights reserved.\n',
-     '17.4'),
-    # Output when LLVM-enabled
-    ('pgcc-llvm 18.4-0 LLVM 64-bit target on x86-64 Linux -tp haswell\n'
-     'PGI Compilers and Tools\n'
-     'Copyright (c) 2018, NVIDIA CORPORATION.  All rights reserved.\n',
-     '18.4')
-])
-def test_pgi_version_detection(version_str, expected_version):
-    version = spack.compilers.pgi.Pgi.extract_version_from_output(version_str)
-    assert version == expected_version
-
-
-@pytest.mark.parametrize('version_str,expected_version', [
-    ('IBM XL C/C++ for Linux, V11.1 (5724-X14)\n'
-     'Version: 11.01.0000.0000\n', '11.1'),
-    ('IBM XL Fortran for Linux, V13.1 (5724-X16)\n'
-     'Version: 13.01.0000.0000\n', '13.1'),
-    ('IBM XL C/C++ for AIX, V11.1 (5724-X13)\n'
-     'Version: 11.01.0000.0009\n', '11.1'),
-    ('IBM XL C/C++ Advanced Edition for Blue Gene/P, V9.0\n'
-     'Version: 09.00.0000.0017\n', '9.0')
-])
-def test_xl_version_detection(version_str, expected_version):
-    version = spack.compilers.xl.Xl.extract_version_from_output(version_str)
-    assert version == expected_version
-
-    version = spack.compilers.xl_r.XlR.extract_version_from_output(version_str)
-    assert version == expected_version
-
-
 @pytest.mark.parametrize('compiler_spec,expected_result', [
     ('gcc@4.7.2', False), ('clang@3.3', False), ('clang@8.0.0', True)
 ])
```

Comment:
This test and a few others like `test_get_compiler_link_paths_no_path` also make sense to move IMO given that this testing module was renamed to `flags.py`

Reply:
@scheibelp Further separating tests on flags is more challenging than expected, due to how things here are mocked up without using fixtures. In the meanwhile I renamed the file `basics.py` (let me know if you have some better name). Let me know if you want me to proceed with a more substantial refactoring that will likely involve move `MockCompiler` and similar functions to a `conftest.py` file in the `compilers` directory  and make them fixtures.